src/measure_theory/covering/liminf_limsup.lean

```diff
@@ -0,0 +1,194 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import measure_theory.covering.density_theorem
+
+/-!
+# Liminf, limsup, and doubling measures.
+
+This file is a place to collect lemmas about liminf and limsup for subsets of a metric space
+carrying a doubling measure.
+
+## Main results:
+
+ * `blimsup_cthickening_mul_ae_eq`: the limsup of the closed thickening of a sequence of compact
+   subsets is unchanged if the sequence of distances is multiplied by a positive scale factor.
+
+-/
+
+open set filter metric measure_theory
+open_locale nnreal ennreal topological_space
+
+variables {α : Type*} [metric_space α] [proper_space α] [measurable_space α] [borel_space α]
+variables (μ : measure α) [is_locally_finite_measure μ] [is_doubling_measure μ]
+
+/-- This is really an auxiliary result en route to `blimsup_cthickening_mul_ae_eq`.
+
+NB: The `set : α` type ascription is present because of issue #16932 on GitHub. -/
+lemma blimsup_cthickening_ae_le_of_eventually_mul_le
```

Comment:
I am not convinced that `hs` is necessary for the result, since the cthickening of a set or its closure should (essentially?) be the same...

Reply:
You are of course absolutely right. Thanks, I have pushed a commit which drops this hypothesis.